7_react/react1/week2/todo-app/src/todoItem/TodoItem.jsx

```diff
@@ -0,0 +1,29 @@
+import "./todoitem.css";
+import { useState } from "react";
+
+export const TodoItem = (props) => {
+  const [itemStatus, setItemStatus] = useState(props.status);
+
+  function toggleStatus(e) {
+    setItemStatus(!itemStatus);
+  }
```

Comment:
Have you noticed that after checking the item, if you interact with your list (add or delete something) the checkmark disappears? That is because your original 'todos' array lives in the TodoList component and TodoItem can not transfer data upstream to mutate it, so if a re-render is triggered you fallback to an array in the TodoList. To be able to change the item status in your todos array you will need to repeat what you have succesfully done with the 'onDeleteClick' function - have the editing function in the same file as your data.
`  const toggleItemStatus = (id) =>  setTodoList(todoList.map(todo => todo.id === id ? { ...todo, status: !todo.status } :  todo));`
Adding function like this to the TodoList component, and when passing it the same way as you've done with 'onDeleteClick' will do the trick 😉